tests/Integration/Tolerant/Indexer/MemberIndexerTest.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace Phpactor\Indexer\Tests\Integration\Tolerant\Indexer;
+
+use Generator;
+use Phpactor\Indexer\Adapter\Tolerant\Indexer\MemberIndexer;
+use Phpactor\Indexer\Adapter\Tolerant\TolerantIndexBuilder;
+use Phpactor\Indexer\Model\Indexer;
+use Phpactor\Indexer\Model\MemberReference;
+use Phpactor\Indexer\Model\Record\FileRecord;
+use Phpactor\Indexer\Model\Record\MemberRecord;
+use Phpactor\Indexer\Tests\Integration\Tolerant\TolerantIndexerTestCase;
+
+class MemberIndexerTest extends TolerantIndexerTestCase
+{
+    public function testRemovesIncomingReferences(): void
+    {
+        $indexer = new MemberIndexer();
+        $index = $this->createIndex();
+        $record1 = $index->get(MemberRecord::fromMemberReference(MemberReference::create('method', 'Foobar', 'bar')));
+        $record2 = $index->get(MemberRecord::fromMemberReference(MemberReference::create('method', 'Foobar', 'bar')));
+
+        $this->assertRemovesIncomingReferences($indexer, $index, $record1, $record2);
+    }
+
+    /**
+     * @dataProvider provideStaticAccess
+     * @dataProvider provideInstanceAccess
+     */
+    public function testMembers(string $manifest, MemberReference $memberReference, int $expectedCount, int $expectedResolvedCount): void
+    {
+        $this->workspace()->reset();
+        $this->workspace()->loadManifest($manifest);
+        $index = $this->createIndex();
+        $indexBuilder = new TolerantIndexBuilder($index, [
+            new MemberIndexer()
+        ]);
+        $fileList = $this->fileListProvider('src');
+        $indexer = new Indexer($indexBuilder, $index, $fileList);
+        $indexer->getJob()->run();
+
+        $memberRecord = $index->get(MemberRecord::fromMemberReference($memberReference));
+        assert($memberRecord instanceof MemberRecord);
+
+        $positionedReferences = [];
+        $resolvedPositionedReferences = [];
+        foreach ($memberRecord->references() as $reference) {
+            $fileRecord = $index->get(FileRecord::fromPath($reference));
+            assert($fileRecord instanceof FileRecord);
+            foreach ($fileRecord->referencesTo($memberRecord) as $positionedReference) {
+                $positionedReferences[] = $positionedReference;
+            }
+            foreach ($fileRecord->referencesTo($memberRecord)->forContainerType($memberReference->containerFqn()) as $positionedReference) {
+                $resolvedPositionedReferences[] = $positionedReference;
+            }
+        }
+
+        self::assertCount($expectedCount, $positionedReferences);
+        self::assertCount($expectedResolvedCount, $resolvedPositionedReferences);
```

Comment:
We should refactor to test each indexer in isolation and remove the monolithic "implementation" test (as we now have only one implementation).

Extract the common stuff here into an abstract method as necesssray,